test/src/fixtures.spec.ts

```diff
@@ -86,31 +82,43 @@ describe('Fixtures', function () {
       puppeteerPath,
       JSON.stringify(options),
     ]);
-    let wsEndPointCallback: (value: string) => void;
-    const wsEndPointPromise = new Promise<string>(x => {
-      wsEndPointCallback = x;
-    });
-    let output = '';
-    res.stdout.on('data', data => {
-      output += data;
-      if (output.indexOf('\n')) {
-        wsEndPointCallback(output.substring(0, output.indexOf('\n')));
+    let killed = false;
+    function killProcess() {
+      if (killed) {
+        return;
       }
-    });
-    const browser = await puppeteer.connect({
-      browserWSEndpoint: await wsEndPointPromise,
-    });
-    const promises = [
-      waitEvent(browser, 'disconnected'),
-      new Promise(resolve => {
-        res.on('close', resolve);
-      }),
-    ];
-    if (process.platform === 'win32') {
-      execSync(`taskkill /pid ${res.pid} /T /F`);
-    } else {
-      process.kill(res.pid!);
+      if (process.platform === 'win32') {
+        execSync(`taskkill /pid ${res.pid} /T /F`);
+      } else {
+        process.kill(res.pid!);
+      }
+      killed = true;
+    }
+    try {
+      let wsEndPointCallback: (value: string) => void;
+      const wsEndPointPromise = new Promise<string>(x => {
+        wsEndPointCallback = x;
+      });
+      let output = '';
+      res.stdout.on('data', data => {
+        output += data;
+        if (output.indexOf('\n')) {
+          wsEndPointCallback(output.substring(0, output.indexOf('\n')));
+        }
+      });
+      const browser = await puppeteer.connect({
+        browserWSEndpoint: await wsEndPointPromise,
+      });
+      const promises = [
+        waitEvent(browser, 'disconnected'),
+        new Promise(resolve => {
+          res.on('close', resolve);
+        }),
+      ];
+      killProcess();
```

Comment:
Do we need this here and in finally?

Reply:
Yes, as we want to close it externally to test the that we throw correct error from Puppeteer if the proccess closes.
In the finally we want to handle time that some error occurred before the first `killProcess();`